systemtest/src/main/java/io/strimzi/systemtest/utils/FileUtils.java

```diff
@@ -28,134 +28,82 @@
 
     private FileUtils() { }
 
-    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
-    public static File downloadAndUnzip(String url) {
-        File dir = null;
-        FileOutputStream fout = null;
-        ZipInputStream zin = null;
-        try {
-            InputStream bais = (InputStream) URI.create(url).toURL().openConnection().getContent();
-            dir = Files.createTempDirectory(FileUtils.class.getName()).toFile();
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE")
+    public static File downloadAndUnzip(String url) throws IOException {
+        File dir = Files.createTempDirectory(FileUtils.class.getName()).toFile();
+
+        try (InputStream bais = (InputStream) URI.create(url).toURL().openConnection().getContent();
+            ZipInputStream zin = new ZipInputStream(bais)) {
+
             dir.deleteOnExit();
-            zin = new ZipInputStream(bais);
+
             ZipEntry entry = zin.getNextEntry();
             byte[] buffer = new byte[8 * 1024];
             int len;
             while (entry != null) {
                 File file = new File(dir, entry.getName());
-                if (entry.isDirectory()) {
-                    file.mkdirs();
-                } else {
-                    fout = new FileOutputStream(file);
-                    while ((len = zin.read(buffer)) != -1) {
-                        fout.write(buffer, 0, len);
+                boolean isDirectoryCreated = file.exists();
```

Comment:
`file` is a filename in a temp directory you just created, so it wouldn't normally exist. I think you want something like this:
```java
if (entry.isDirectory()) {
    if (file.exists()) {
        if (!file.isDirectory()) {
            throw new IOException("Malformed zip file");
        }
    } else {
        if (!file.mkdirs()) {
            throw new IOException("Could not create directory " + file);
        }
    }
} else {
    file.getParentFile().mkdirs(); // create parent; in case zip file is malformed
    try (FileOutputStream fout = new FileOutputStream(file)) {
    while ((len = zin.read(buffer)) != -1) {
        fout.write(buffer, 0, len);
    }
}
```
You could have similar logic for the 2nd `mkdirs()` as for the first (and maybe factor into a `maybeCreateDirectory()` method).